addons/link_tracker/models/mail_render_mixin.py

```diff
@@ -36,22 +36,22 @@ def _shorten_links(self, html, link_tracker_vals, blacklist=None, base_url=None)
         """
         base_url = base_url or self.env['ir.config_parameter'].sudo().get_param('web.base.url')
         short_schema = base_url + '/r/'
-        for match in set(re.findall(tools.HTML_TAG_URL_REGEX, html)):
-            long_url = match[1]
+        for match in set(re.findall(tools.HTML_TAG_URL_WITH_IMAGE_ALT_REGEX, html)):
+            long_url = match[2]
             # Don't shorten already-shortened links
             if long_url.startswith(short_schema):
                 continue
             # Don't shorten urls present in blacklist (aka to skip list)
             if blacklist and any(s in long_url for s in blacklist):
                 continue
-            label = (match[3] or '').strip()
+            label = (match[8] or match[6] or match[5]).strip()
```

Comment:
to be sure it will never crash (not 100% sure it can crash :thinking: )
```suggestion
            label = (match[8] or match[6] or match[5] or '').strip()
```